### PR TITLE
SNOW-1011775: Renaming 'containers' workload to 'spcs'. Moving 'image-registry' under 'spcs'.

### PR DESCRIPTION
**Pull request by sfc-gh-davwang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Renamed the 'containers' command group to 'spcs' (Snowpark Container Services). Also moved the registry command group under 'spcs'.

I have another branch rebased off of Tomasz's PR that enables SPCS integration tests (https://github.com/Snowflake-Labs/snowcli/pull/684). The rebased branch passes the SPCS integration tests: https://github.com/Snowflake-Labs/snowcli/pull/688.

### CODE REVIEW

File: src/snowflake/cli/plugins/spcs/__init__.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-from snowflake.cli.plugins.containers.services.commands import app as services_app
+from snowflake.cli.plugins.spcs.jobs.commands import app as jobs_app
+from snowflake.cli.plugins.spcs.registry.commands import app as registry_app
+from snowflake.cli.plugins.spcs.services.commands import app as services_app
 
 app = typer.Typer(
-    name="containers",
+    name="spcs",
     context_settings=DEFAULT_CONTEXT_SETTINGS,
     help="Manages Snowpark services, pools and jobs.",
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/spcs/__init__.py:13]: With moving registry here, should we update it? I'm fine with current state, just confirming

File: tests/spcs/test_jobs.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def test_create_job(mock_connector, runner, mock_ctx):
         "EXECUTE SERVICE\n"
         "IN COMPUTE POOL testPool\n"
         "FROM SPECIFICATION $$\n"
-        '{"spec": {"containers": [{"name": "main", "image": "public.ecr.aws/myrepo:latest"}]}}\n'
+        '{"spec": {"spcs": [{"name": "main", "image": "public.ecr.aws/myrepo:latest"}]}}\n'
```
Comment (sfc-gh-turbaszek) [inline tests/spcs/test_jobs.py:38]: That's not intended change I think

File: tests/spcs/test_services.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def test_create_service(mock_execute_schema_query, other_directory):
     expected_query = (
         "CREATE SERVICE IF NOT EXISTS test_service "
         "IN COMPUTE POOL test_pool "
-        'FROM SPECIFICATION $$ {"spec": {"containers": [{"name": "cloudbeaver", "image": '
+        'FROM SPECIFICATION $$ {"spec": {"spcs": [{"name": "cloudbeaver", "image": '
```
Comment (sfc-gh-turbaszek) [inline tests/spcs/test_services.py:46]: That's not intended change I think

File: tests/spcs/test_services.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def test_create_service(mock_execute_schema_query, other_directory):
         dedent(
             """
     spec:
-        containers:
+        spcs:
```
Comment (sfc-gh-turbaszek) [inline tests/spcs/test_services.py:25]: That's not intended change I think

**Comment by sfc-gh-turbaszek** [comment]
Overall this looks good to me. There are few renames to be fixed and E2E tests to be adjusted.

Have you run integration tests for spcs? We don't have them enabled on CI/CI because SPCS are not yet rolled out on our account.

File: src/snowflake/cli/plugins/spcs/__init__.py (hunk 1 above)
Comment (sfc-gh-davwang) [reply src/snowflake/cli/plugins/spcs/__init__.py]: Good point, I've updated the note to include image registries and repositories for clarity.

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: tests_e2e/__snapshots__/test_installation.ambr [hunk 5]
```diff
@@ -41,14 +41,14 @@
...
-  │ object         Manages Snowflake objects like warehouses and stages          │
-  │ registry       Manages Snowpark registries.                                  │
-  │ snowpark       Manage procedures and functions.                              │
-  │ sql            Executes Snowflake query.                                     │
-  │ streamlit      Manages Streamlit in Snowflake.                               │
+  │ app         Manage Native Apps in Snowflake                                  │
+  │ connection  Manages connections to Snowflake.                                │
+  │ object      Manages Snowflake objects like warehouses and stages             │
+  │ snowpark    Manage procedures and functions.                                 │
+  │ spcs        Manages Snowpark services, pools, jobs, image registries, and    │
```
Comment (sfc-gh-yiwang) [inline tests_e2e/__snapshots__/test_installation.ambr:48]: nit: use the spcs full name here - Manages Snowpark Container Services

**Pull request merged**